packages/graphql-hooks/src/middlewares/apqMiddleware.js

```diff
@@ -0,0 +1,42 @@
+async function sha256(data) {
+  // MOCK
+  return data
+    .split('')
+    .reduce((acc, ch) => acc + ch.charCodeAt(0), '')
+    .substr(0, 8)
+}
+
+/**
+ * AutomaticPersistedQueryMiddleware - must be last in the middleware list
+ * @param {function} makeRequest
+ * @returns Promise<object>
+ */
+const APQMiddleware = async ({ operation, client, resolve, reject }, next) => {
+  try {
+    // Add SHA256 of a query
+    operation = { ...operation, hash: await sha256(operation.query) }
```

Comment:
If we want something solid (and proved to be working) we can use something like this:
https://github.com/rickbergfalk/postgrator/blob/master/lib/utils.js#L39
It is true that literally any change will make an impact, but that might be ok?

Reply:
Two concerns: 1) I assumed APQ needs sha256, 2) need to check if the crypto lib is present in the browser. I'll take a look in the next PR, thanks 👍